bee-api/bee-rest-api/src/endpoints/routes/api/v1/messages/mod.rs

```diff
@@ -0,0 +1,245 @@
+                    Ok(data) => data,
+                    Err(e) => json!(ErrorBody::new(DefaultErrorResponse {
+                        code: "400".to_string(),
+                        message: format!("could not convert message. {}", e)
+                    })),
+                },
+            ),
+            None => Json(json!(ErrorBody::new(DefaultErrorResponse {
+                code: "400".to_string(),
+                message: "could find message.".to_string()
+            }))),
+        },
+        Err(e) => Json(json!(ErrorBody::new(DefaultErrorResponse {
+            code: "400".to_string(),
+            message: format!("could not fetch message from storage. {}", e)
+        }))),
+    }
+}
+
+pub async fn get_id_metadata_handler(
+    Path(messageid): Path<String>,
+    Extension(app_storage): Extension<Arc<AppStorage>>,
+) -> Json<Value> {
+    match Fetch::<MessageId, Message>::fetch(
+        &*(app_storage.storage.lock().unwrap()),
+        match &MessageId::from_str(&messageid) {
+            Ok(message_id) => message_id,
+            Err(e) => {
+                return Json(json!(ErrorBody::new(DefaultErrorResponse {
+                    code: "400".to_string(),
+                    message: format!("could not parse message ID. {}", e)
+                })));
+            }
+        },
+    ) {
+        Ok(message) => match message {
+            Some(message) => Json(
+                match serde_json::to_value(&SuccessBody::new({
+                    let message_dto = MessageDto::from(&message);
+                    MessageMetadataResponse {
+                        message_id: message.id().to_string(),
+                        parent_message_ids: message_dto.parents.iter().map(|p| p.message_id.clone()).collect(),
+                    }
+                })) {
+                    Ok(data) => data,
+                    Err(e) => json!(ErrorBody::new(DefaultErrorResponse {
+                        code: "400".to_string(),
+                        message: format!("could not convert message. {}", e)
+                    })),
+                },
+            ),
+            None => Json(json!(ErrorBody::new(DefaultErrorResponse {
+                code: "400".to_string(),
+                message: "could find message.".to_string()
+            }))),
+        },
+        Err(e) => Json(json!(ErrorBody::new(DefaultErrorResponse {
+            code: "400".to_string(),
+            message: format!("could not fetch message from storage. {}", e)
+        }))),
+    }
+}
+
+pub async fn get_id_raw_handler(
+    Path(messageid): Path<String>,
+    Extension(app_storage): Extension<Arc<AppStorage>>,
+) -> Json<Value> {
+    match Fetch::<MessageId, Message>::fetch(
+        &*(app_storage.storage.lock().unwrap()),
+        match &MessageId::from_str(&messageid) {
+            Ok(message_id) => message_id,
+            Err(e) => {
+                return Json(json!(ErrorBody::new(DefaultErrorResponse {
```

Comment:
Instead of writing every time `Json(json!(ErrorBody::new(DefaultErrorResponse { ... } ` do you think it makes sense to have a `fn error(code, message) -> Json<Value>` function which will be called instead?